smart-contracts/contracts/airdrop/src/contract.rs

```diff
@@ -0,0 +1,194 @@
+use cosmwasm_std::{
+    entry_point, to_binary, Attribute, Binary, Deps, DepsMut, Env, Event, MessageInfo, Reply,
+    Response, StdResult, Uint128,
+};
+use cw2::set_contract_version;
+use cw20_base::msg::MigrateMsg;
+
+use crate::admin::{
+    execute_add_users, execute_remove_users, execute_set_users, execute_update_airdrop_config,
+    execute_withdraw_funds,
+};
+use crate::error::AirdropErrors;
+use crate::helpers::is_contract_admin;
+use crate::msg::{AdminExecuteMsg, ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::query::{query_config, query_contract_state, query_sanity_check, query_user};
+use crate::reply::handle_reply;
+use crate::state::{AIRDROP_CONFIG, REPLY_MAP};
+use crate::users::execute_claim;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "quasar_airdrop";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Instantiates the airdrop contract with the provided configuration.
+///
+/// # Arguments
+///
+/// * `deps` - Dependencies to access storage and external data.
+/// * `_env` - Environment information, not used in this function.
+/// * `_info` - Message sender's information, not used in this function.
+/// * `msg` - Instantiate message containing the airdrop configuration.
+///
+/// # Errors
+///
+/// Returns an error if the airdrop configuration is invalid, specifically if start height,
+/// end height, and total claimed are not set to zero.
+///
+/// # Returns
+///
+/// Returns a response indicating the success of contract instantiation and includes attributes
+/// describing the airdrop configuration.
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, AirdropErrors> {
+    // Set the contract version in storage
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // Check if the contract should not be instantiated due to an invalid airdrop window
+    if msg.config.start_height != 0
```

Comment:
&& shound'nt this be || if you want to initialise the contract with zero values of start height and end height. And msg.config.total_claimed is zero.